rlclientlib/utility/context_helper.cc

```diff
@@ -144,4 +145,35 @@ namespace reinforcement_learning { namespace utility {
     }
     return error_code::success;
   }
+
+  int get_slot_ids_or_add_string_index(const char* context, const ContextInfo::index_vector_t& slots, std::vector<std::string>& slot_ids, i_trace* trace, api_status* status)

```

Comment:
[nit] I think get_slot_ids is enough for the name. If you want to elaborate I would change the second part to convert/generate_from_index

Reply:
I tend to prefer elaborate names so opted for changing to: `or_generate_from_index`